src/test/java/com/dawn/jat/illuminati/post/service/PostServiceTest.java

```diff
@@ -114,45 +126,55 @@ void create() {
                 .thenReturn(postEntity);
         Mockito.when(postRepository.savePost(any(PostEntity.class)))
                 .thenReturn(postEntity);
-        assertThat(postService.create(postDto), is(postEntity));
+        Mockito.when(modelMapper.map(any(), any())).thenReturn(postDto);
+        assertThat(postService.createPost(postDto), is(postDto));
         Mockito.verify(postRepository, Mockito.times(1)).savePost(postEntity);
     }
 
     @Test
     void savePost_WithDto_SuccessfullySaveDto() {
-
-        PostEntity expectedPost = new PostEntity("how-to-apply-agile-methodology",
+        PostEntity expectedPost = new PostEntity(
+                "how-to-apply-agile-methodology",
                 "How to apply Agile methodology new",
                 "How to apply Agile methodology new",
                 "01/01/2020 new",
                 new ArrayList<>(Arrays.asList("System Design", "OOP")),
-                "Li Li new");
+                "Li Li new",
+                "");
         expectedPost.setContent("new content");
-        Mockito.when(postRepository.findById(postDto.getId()))
-                .thenReturn(Optional.of(postEntity));
+        String id = postDto.getId();
+        assertNotNull(id);
+
+        Mockito.when(postRepository.existsById(id))
+                .thenReturn(true);
+
         Mockito.when(postRepository.findById(postDto.getId()))
                 .thenReturn(Optional.of(postEntity));
 
-        postService.save(postDto);
+        Mockito.when(converter.convertPostDtoToEntity(eq(postDto), any(PostEntity.class)))
```

Comment:
should we mock this one? Because the convert logic doesn't have anything relevant to any storage or database 🤔 And the logic of this function is not complicated for us to do the mock